### PR TITLE
Examples, basic, consul seed and monitoring

### LINKED ISSUE #19: Provide examples
## Is your feature request related to a problem? Please describe

We should provide a examples folder in order to show how to use harvester.

## Describe the solution

Create some examples for the usage of harvester.

### PR DESCRIPTION
**Pull request by dbaltas**
Signed-off-by: Dimitris Baltas <dimitris.baltas@thebeat.co>

## Which problem is this PR solving?

Examples for usage closes #19 

## Short description of the changes
Add example for
* usage with ENV vars
* usage with consul

### CODE REVIEW

File: examples/01_basic/main.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	attrs := ConfigAttrs{}
+
+	cfg, err := config.New(&attrs)
+	if err != nil {
+		fmt.Printf("Oops, something went wrong creating config: %v", err)
+	}
+
+	printAttrs(attrs)
+
+	s := seed.New()
```
Comment (mantzas) [inline examples/01_basic/main.go]: You should use harvester and not the individual components for config and seed.

File: examples/02_consul/main.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	// PUT a new KV pair
+	p := &api.KVPair{Key: "harvester/example_02/consul_var", Value: []byte("bar")}
+	_, err = kv.Put(p, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	attrs := ConfigAttrs{}
+
+	cfg, err := config.New(&attrs)
```
Comment (mantzas) [inline examples/02_consul/main.go]: You should use harvester and not the individual components for config and seed.

File: examples/02_consul/main.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/taxibeat/harvester/config"
+	"github.com/taxibeat/harvester/seed"
+	"github.com/taxibeat/harvester/seed/consul"
+)
+
+type fields struct {
+	consulParam *seed.Param
+}
+
+// ConfigAttrs sample config struct
```
Comment (mantzas) [inline examples/02_consul/main.go]: this one should not be exported.

Comment (dbaltas) [reply examples/02_consul/main.go]: 👍

File: examples/02_consul/main.go (hunk 2 above)
Comment (dbaltas) [reply examples/02_consul/main.go]: 👍

File: examples/01_basic/main.go (hunk 1 above)
Comment (dbaltas) [reply examples/01_basic/main.go]: 👍

**Comment by dbaltas** [comment]
@mantzas I prefer 3 examples as in this PR, in the case that Harvester provides value even without consul support.

**Comment by dbaltas** [comment]
@mantzas This PR was moved to wip because I had trouble seeing an active modification take effect.
I tried modifications in two ways
1. by calling `example03::seedConsulVars(consulConfig, "bazz")`
2. by  updating consul through curl `curl --request PUT --data bazz http://127.0.0.1:8500/v1/kv/harvester/example_03/consul_var`
None of the above did the trick

**Comment by mantzas** [comment]
> @mantzas This PR was moved to wip because I had trouble seeing an active modification take effect.
> I tried modifications in two ways
> 
> 1. by calling `example03::seedConsulVars(consulConfig, "bazz")`
> 2. by  updating consul through curl `curl --request PUT --data bazz http://127.0.0.1:8500/v1/kv/harvester/example_03/consul_var`
>    None of the above did the trick

check out https://github.com/taxibeat/harvester/pull/14 which contains a test that i had in the codebase but removed it cbecause it generates a data race, which is actually expected but in order to remove this i have to refactor not only code but the initial user config in order to support proper locking...

**Pull request merged**